src/utils/getPropertyViaDotNotation.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Retrieves a nested property from an object using a dot-notation string.
+ *
+ * @template T - The type of the object.
+ * @param propertyName - The dot-notation string representing the property path.
+ * @param object - The object from which to retrieve the property.
+ * @returns The value of the nested property or undefined if it doesn't exist.
+ */
+export default function getPropertyViaDotNotation<T extends Record<string, any>>(
+  propertyName: string,
+  object: T
```

Comment:
Wouldn't this be a `Record<string, any>`

Reply:
Yeap, what is defined on the generic. T is a Record<string,any>